common/persistence/sql/sqlplugin/mysql/plugin.go

```diff
@@ -84,6 +91,13 @@ func (p *plugin) CreateAdminDB(cfg *config.SQL) (sqlplugin.AdminDB, error) {
 // SQL database and the object can be used to perform CRUD operations on
 // the tables in the database
 func (p *plugin) createDBConnection(cfg *config.SQL) (*sqlx.DB, error) {
+	if cfg.TLS != nil && cfg.TLS.Enabled {
```

Comment:
do this check within the registerTLSConfig. If tls is not enabled, simply return nil from registerTLSConfig